kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/paths/paths.kt

```diff
@@ -203,9 +208,9 @@ infix fun Path.shouldNotContainFileDeep(name: String) = this shouldNot containFi
 fun containFileDeep(name: String): Matcher<Path> = object : Matcher<Path> {
 
    private fun fileExists(dir: Path): Boolean {
-      val contents = Files.list(dir).toList()
-      val (dirs, files) = contents.partition { Files.isDirectory(it) }
-      return files.map { it.fileName.toString() }.contains(name) || dirs.any(::fileExists)
+      val contents = dir.toFile().listFiles()
+      val (dirs, files) = contents.partition { it.isDirectory }
+      return files.map { it.name.toString() }.contains(name) || dirs.any { fileExists(it.toPath()) }
```

Comment:
Rather than converting Paths to Files, I would prefer either using the Java NIO functions (so revert this change), or use Kotlin stdlib utils so using Paths is easier

```suggestion
      val contents = dir.listDirectoryEntries()
      val (dirs, files) = contents.partition { it.isDirectory() }
      return files.map { it.name }.contains(name) || dirs.any { fileExists(it) }
```

(This change also requires `import kotlin.io.path.*`)

Reply:
I don't know why but I can't get the original to compile anymore.